### PR TITLE
Add java docs, also rename some classes and functions

### PR DESCRIPTION
**Pull request by kestivvi**
I have added a milion java docs to classes and functions I'd created. Maybe I will wait with merging it until the #44 is merged, because it was renaming some classes as well and then I improve this PR.

I should've also add TODOs to functions that are possible to be unit tested.
- [x] Added TODOs

### CODE REVIEW

**Comment by kestivvi** [comment]
I should've also add TODOs to functions that are possible to be unit tested.

- [ ] Added TODOs

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
     public JsonIsNullException(Throwable cause) {
         super(cause);
     }
 
+    /**
+     * Constructs a new JsonIsNullException with the specified detail message and cause.
+     *
+     * @param message the detail message
+     * @param cause the cause
+     */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: I think that commenting this part of the code is useless.

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java [hunk 2]
```diff
@@ -1,17 +1,36 @@
 package com.github.splendor_mobile_game.websocket.utils.json.exceptions;
 
+/** Exception thrown when a JSON object is null. */
 public class JsonIsNullException extends JsonParserException {
+
+    /** Constructs a new JsonIsNullException with no message. */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: ? xd This type of comments makes code harder to read, not easier

**Review by Vertonowsky — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java (hunk 2 above)
Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: So what are you proposing? To remove all java docs from the all exceptions?

Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: yeah. Also we don't need comments on basic functions with just 1/2 lines of code. It's just making a bush from our code

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java (hunk 1 above)
Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: Same here. Function is so simple, that there shouldn't be any comments.

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java (hunk 2 above)
Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: I wonder if our lecturer wouldn't require everything to be documented. Even the simplest functions.

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: However, if you would like, I can remove java docs from exceptions and the simplest functions. What's the decision, Mr. Programmer?

Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: We can wait untill tomorrow and we will ask about the documentation. But in my opinion not always good = a lot.

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: I agree with you

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: Lecturer has said that we can remove java docs from constructors in exceptions if they are default implementations, leaving only general java doc about exception class as a whole. He said also we can remove java docs from getter and setters and other functions that contain only one line or two and the method name says everything you want to know.

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/utils/json/exceptions/JsonIsNullException.java]: I've just implemented those changes in my java docs.

File: src/main/java/com/github/splendor_mobile_game/websocket/communication/WebSocketConnectionChecker.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
  */
 public class WebSocketConnectionChecker implements Runnable {
 
-    /** The WebSocket connection. */
+    /** The WebSocket connection this instance is monitoring. */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/communication/WebSocketConnectionChecker.java]: It says nothing to me. Refactor or remove. (i prefer removing)

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java [hunk 4]
```diff
@@ -1,9 +1,19 @@
 package com.github.splendor_mobile_game.websocket.handlers;
 
+/** Represents a message to be sent to a receiver identified by their websocket hashcode. */
 public class Message {
+
+    /** The websocket hashcode of the receiver of the message. */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java]: "Each connection between User and Server is identified by some ID. `receiverHashcode` is the ID of the connection.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java [hunk 5]
```diff
@@ -1,9 +1,19 @@
...
     private int receiverHashcode;
+
+    /** The content of the message. */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java]: unnecessary

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java [hunk 6]
```diff
@@ -1,9 +1,19 @@
...
     private String message;
-    
+
+    /**
+     * Constructs a new Message object with the given receiver websocket hashcode and message content.
+     * @param receiverHashcode the websocket hashcode of the receiver of the message
+     * @param message the content of the message
+     */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/Message.java]: unnecessary

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+public abstract class ConnectionChecker {
+
+    /** The WebSocket connection to check. */
+    protected WebSocket connection;
+
+    /**
+     * Creates a new ConnectionChecker instance.
+     *
+     * @param connection the WebSocket connection to check
+     */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java]: unnecessary

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    /**
+     * Called when a connection check is performed.
+     *
+     * @param timeSinceLastPongMs the time since the last pong message was received, in milliseconds
+     */
+    public abstract void onConnectionCheck(Long timeSinceLastPongMs);
+
+    /** Called when the WebSocket connection is closed. */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java]: unnecessary

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    public ConnectionChecker(WebSocket connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * Called when a connection check is performed.
+     *
+     * @param timeSinceLastPongMs the time since the last pong message was received, in milliseconds
+     */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/connection/ConnectionChecker.java]: unnecessary

File: src/main/java/com/github/splendor_mobile_game/websocket/utils/CustomException.java [hunk 10]
```diff
@@ -19,11 +25,21 @@ public CustomException(String message, Throwable cause) {
         super(message, cause);
     }
 
+    /**
+     * Returns a string representation of this custom exception, including the message and cause.
+     *
+     * @return a string representation of this custom exception
+     */
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/utils/CustomException.java]: unnecessary

**Review by Vertonowsky — CHANGES_REQUESTED** [review]

**Comment by kestivvi** [comment]
I've followed your lead @Vertonowsky.

**Review by Vertonowsky — APPROVED** [review]

**Pull request merged**